feat(kong 1.1) rewrote docs for Kong 1.1

hutchic opened the pull request:
Merge https://github.com/Kong/kong-dist-kubernetes/pull/73 first

hutchic committed:
feat(kong 1.1) rewrote docs for Kong 1.1

hutchic commented on README.md:
```diff
@@ -20,51 +20,20 @@ For inspecting the running system inside of minikube with docker, run:
+### Run Kong on k8s
+git clone https://github.com/Kong/kong-dist-kubernetes.git
+cd kong-dist-kubernetes
+git checkout feat/kong-1.1
```
todo: remove this after https://github.com/Kong/kong-dist-kubernetes/pull/73 merges

james-callahan commented on README.md:
```diff
@@ -20,51 +20,20 @@ For inspecting the running system inside of minikube with docker, run:
-KONG_ADMIN_URL=$(minikube service -n kong kong-control-plane --url | head -n 1)
-curl $KONG_ADMIN_URL/plugins -d name=kubernetes-sidecar-injector
+
+kubectl port-forward -n kong svc/kong-control-plane 8001:80
```
Doesn't this start a long running process?

hutchic replied:
It does which I think is fine?

james-callahan replied:
So doesn't the example just sit here doing nothing?
You're not putting it into the background or anything (and if you want it to, make sure you add clean up code to kill the background process when the example is over)

hutchic replied:
I habitually use multiple terminal tabs but have a habit of not documenting that (whoops). I tweaked the line but plan is a fully polished version ends up on docs.konghq.com utilizing bookinfo.yaml

james-callahan commented on kong/plugins/kubernetes-sidecar-injector/typedefs.lua:
```diff
@@ -1,222 +0,0 @@
-local Schema = require "kong.db.schema"
```
Why is all this code getting deleted?

hutchic replied:
weird pretty sure it wasn't when this PR was a draft 🤔

hutchic committed:
doc(k8s) adjust the sample k8s documentation

docs(README) pr feedback adjustments

docs(k8s) remove the branch specifier just use master

hbagdi commented on README.md:
````diff
@@ -95,6 +64,21 @@ webhooks:
 EOF
 ```
 
+Going forward any pod's that start get a Kong sidecar automatically injected which all data from the containers
````
Please respect the 80 character column length here.

hbagdi commented on kong/plugins/kubernetes-sidecar-injector/handler.lua:
```diff
@@ -1,10 +1,7 @@
+  VERSION = "0.2.0",
+  -- priority doesn't matter, just need to pick something unique for kong tests
+  PRIORITY = 1006,
+}
```
Is this code needed?
We ship the sidecar injector with Kong.

james-callahan replied:
This code *is* what gets shipped with kong.

hbagdi replied:
I see.
It feels odd that we are updating this code in this PR then. Please open a separate PR for this change.

hbagdi replied:
@hutchic Please remove this change and we are good to merge.

hutchic replied:
this matches master and I squashed the commits. I think it only shows up in the github diff because of the comment

james-callahan commented on Dockerfile:
```diff
@@ -1,5 +0,0 @@
-FROM kong
```
With this deleted, how should a developer work on the plugin?

hutchic replied:
I'm curious how do other plugins handle this? I'm fine with whatever but consistency would be nice but that's a someday task probably :)

james-callahan commented on README.md:
```diff
@@ -20,51 +20,20 @@ For inspecting the running system inside of minikube with docker, run:
+git clone https://github.com/Kong/kong-dist-kubernetes.git
+cd kong-dist-kubernetes
+git checkout feat/kong-1.1
+make run_<postgres|cassandra>
```
I don't like needing `make` for this

hutchic replied:
Trying to reduce copy-pasta and https://github.com/Kong/kong-dist-kubernetes/blob/feat/kong-1.1/Makefile#L6-L14 is the logical place to keep all "run Kong on k8s" right now but open to suggestions

james-callahan commented on README.md:
```diff
@@ -20,51 +20,20 @@ For inspecting the running system inside of minikube with docker, run:
+### Run Kong on k8s
+git clone https://github.com/Kong/kong-dist-kubernetes.git
+cd kong-dist-kubernetes
+git checkout feat/kong-1.1
```
How will this get updated in future?

hutchic replied:
It won't it will just use master going forward per my comment 👍

hbagdi replied:
I recommend then merging the other PR first, make the change here and then merge this in.

james-callahan commented on README.md:
```diff
@@ -20,51 +20,20 @@ For inspecting the running system inside of minikube with docker, run:
-KONG_ADMIN_URL=$(minikube service -n kong kong-control-plane --url | head -n 1)
-curl $KONG_ADMIN_URL/plugins -d name=kubernetes-sidecar-injector
+
+kubectl port-forward -n kong svc/kong-control-plane 8001:80 &
```
This needs to be killed somehow.
